test/e2e/operator.go

```diff
@@ -272,3 +276,108 @@ var _ = Describe("ARO Operator - Conditions", func() {
 		Expect(err).NotTo(HaveOccurred())
 	})
 })
+
+var _ = Describe("ARO Operator - Azure NSG Reconciler", func() {
+	var vnetName, location, resourceGroup, workerSubnet, masterSubnet string
+	var dummynsg mgmtnetwork.SecurityGroup
+	ctx := context.Background()
+	enableReconcileNSG := func() {
+		instance, err := clients.AROClusters.AroV1alpha1().Clusters().Get(ctx, arov1alpha1.SingletonClusterName, metav1.GetOptions{})
+		Expect(err).NotTo(HaveOccurred())
+		if !instance.Spec.Features.ReconcileNSGs {
+			instance.Spec.Features.ReconcileNSGs = true
+			_, err = clients.AROClusters.AroV1alpha1().Clusters().Update(ctx, instance, metav1.UpdateOptions{})
+			Expect(err).NotTo(HaveOccurred())
+		}
+	}
+	gatherNetworkInfo := func() {
+		oc, err := clients.OpenshiftClustersv20200430.Get(ctx, vnetResourceGroup, clusterName)
+		Expect(err).NotTo(HaveOccurred())
+		vnetID, _, err := subnet.Split(*oc.OpenShiftClusterProperties.MasterProfile.SubnetID)
+		Expect(err).NotTo(HaveOccurred())
+
+		r, err := azure.ParseResourceID(vnetID)
+		Expect(err).NotTo(HaveOccurred())
+		resourceGroup = r.ResourceGroup
+
+		vnet, err := clients.VirtualNetworks.Get(ctx, resourceGroup, r.ResourceName, "")
+		Expect(err).NotTo(HaveOccurred())
+		location = *vnet.Location
+		vnetName = *vnet.Name
+
+		_, masterSubnet, err = subnet.Split(strings.ToLower(*oc.OpenShiftClusterProperties.MasterProfile.SubnetID))
+		Expect(err).NotTo(HaveOccurred())
+		_, workerSubnet, err = subnet.Split(strings.ToLower(*(*oc.OpenShiftClusterProperties.WorkerProfiles)[0].SubnetID))
+		Expect(err).NotTo(HaveOccurred())
+	}
+	createDummyNSG := func() {
+		dummynsg = mgmtnetwork.SecurityGroup{
+			Location:                      to.StringPtr(location),
+			Name:                          to.StringPtr("dummynsg"),
+			Type:                          to.StringPtr("Microsoft.Network/networkSecurityGroups"),
+			SecurityGroupPropertiesFormat: &mgmtnetwork.SecurityGroupPropertiesFormat{},
+		}
+		err := clients.NetworkSecurityGroups.CreateOrUpdateAndWait(ctx, resourceGroup, "dummynsg", dummynsg)
+		Expect(err).NotTo(HaveOccurred())
+		dummynsgObject, err := clients.NetworkSecurityGroups.Get(ctx, resourceGroup, "dummynsg", "")
+		Expect(err).NotTo(HaveOccurred())
+		dummynsg.ID = dummynsgObject.ID
```

Comment:
Maybe just replace `dummynsg` with a full object?

```suggestion
		dummynsg, err = clients.NetworkSecurityGroups.Get(ctx, resourceGroup, "dummynsg", "")
		Expect(err).NotTo(HaveOccurred())
```